clients/src/test/java/org/apache/kafka/clients/admin/KafkaAdminClientTest.java

```diff
@@ -215,6 +215,41 @@ public void testCreateTopics() throws Exception {
         }
     }
 
+    @Test
+    public void testInvalidTopicNames() throws Exception {
```

Comment:
nit (open to discussion): I tend to prefer smaller test cases when they are possible since it is easier to narrow down the problem. Any reason not to split this into 3 separate cases: one for each API?

Reply:
It's just more boilerplate creating each test.  This particular test actually runs very fast since it's a true unit test (no brokers running), so test time should not be an issue.  I don't feel strongly about it, though.

Reply:
Yeah, I don't feel too strongly in this case. The advantage generally is that the scope of the test case is clearer which makes failures easier to investigate.